bridges/BadDragonBridge.php

```diff
@@ -0,0 +1,417 @@
+<?php
+class BadDragonBridge extends BridgeAbstract {
+	const NAME = 'Bad Dragon Bridge';
+	const URI = 'https://bad-dragon.com/';
+	const CACHE_TIMEOUT = 300; // 5min
+	const DESCRIPTION = 'Returns sales or new clearance items';
+	const MAINTAINER = 'Roliga';
+	const PARAMETERS = array(
+		'Sales' => array(
+		),
+		'Clearance' => array(
+			'ready_made' => array(
+				'name' => 'Ready Made',
+				'type' => 'checkbox'
+			),
+			'flop' => array(
+				'name' => 'Flops',
+				'type' => 'checkbox'
+			),
+			'skus' => array(
+				'name' => 'Products',
+				'exampleValue' => 'chanceflared, crackers',
+				'title' => 'Comma separated list of product SKUs'
+			),
+			'onesize' => array(
+				'name' => 'One-Size',
+				'type' => 'checkbox'
+			),
+			'mini' => array(
+				'name' => 'Mini',
+				'type' => 'checkbox'
+			),
+			'small' => array(
+				'name' => 'Small',
+				'type' => 'checkbox'
+			),
+			'medium' => array(
+				'name' => 'Medium',
+				'type' => 'checkbox'
+			),
+			'large' => array(
+				'name' => 'Large',
+				'type' => 'checkbox'
+			),
+			'extralarge' => array(
+				'name' => 'Extra Large',
+				'type' => 'checkbox'
+			),
+			'category' => array(
+				'name' => 'Category',
+				'type' => 'list',
+				'values' => array(
+					'All' => 'all',
+					'Accessories' => 'accessories',
+					'Merchandise' => 'merchandise',
+					'Dildos' => 'insertable',
+					'Masturbators' => 'penetrable',
+					'Packers' => 'packer',
+					'Lil\' Squirts' => 'shooter',
+					'Lil\' Vibes' => 'vibrator',
+					'Wearables' => 'wearable'
+				),
+				'defaultValue' => 'all',
+			),
+			'soft' => array(
+				'name' => 'Soft Firmness',
+				'type' => 'checkbox'
+			),
+			'med_firm' => array(
+				'name' => 'Medium Firmness',
+				'type' => 'checkbox'
+			),
+			'firm' => array(
+				'name' => 'Firm',
+				'type' => 'checkbox'
+			),
+			'split' => array(
+				'name' => 'Split Firmness',
+				'type' => 'checkbox'
+			),
+			'maxprice' => array(
+				'name' => 'Max Price',
+				'type' => 'number',
+				'required' => true,
+				'defaultValue' => 300
+			),
+			'minprice' => array(
+				'name' => 'Min Price',
+				'type' => 'number',
+				'defaultValue' => 0
+			),
+			'cumtube' => array(
+				'name' => 'Cumtube',
+				'type' => 'checkbox'
+			),
+			'suctionCup' => array(
+				'name' => 'Suction Cup',
+				'type' => 'checkbox'
+			),
+			'noAccessories' => array(
+				'name' => 'No Accessories',
+				'type' => 'checkbox'
+			)
+		)
+	);
+
+	private function setParam($inArr, &$outArr, $param, $strFrom, $strTo = null) {
+		if(isset($inArr[$param]) && in_array($strFrom, $inArr[$param])) {
+			$outArr[($strTo ?: $strFrom)] = 'on';
+		}
+	}
+
+	public function detectParameters($url) {
```

Comment:
What is the point of this function ?
Same for `setParam`, that is only called in this function

Reply:
`detectParameters` is for automatically selecting a bridge from a URL. See wiki [here](https://github.com/RSS-Bridge/rss-bridge/wiki/The-detectParameters-function).

`setParam` sets index `$strFrom` (or `$strTo` if it's set) in `$outArr` to *'on'* if `$inArr[$param]` contains `$strFrom`. It's for translating their shop filter URL parameters into something rss-bridge can use, for example:

```
Array
(
    [type] => Array
        (
            [0] => ready_made
            [1] => flop
        )

    [firmnessValues] => Array
        (
            [0] => medium
        )
)
```
to:
```
Array
(
    [ready_made] => on
    [flop] => on
    [med_firm] => on
)
```
Perhaps there's some nicer way to do that though, or at least a comment explaining `setParam` could be added.

Reply:
Honestly, the function `setParam` is hard to comprehend because the arguments (and the function name) don't add to clarity. It's generally good practice to name variables according to their purpose, not their type.

What do `$inArr`,  `$outArr`, `$param`, `$strFrom` and `$strTo` really represent?

From your explanation I understand that the function should be named `transformQuery` (or similar). It also makes sense to include an example in the documentation for future reference.

Does this work for you?

```php
/** 
 * Adds one element from a multi-value input query to a single-value
 * output query, using the value of the input element as parameter
 * name in the output query and setting the value to 'on'.
 *
 * The parameter name can optionally be changed with $outputParam
 *
 * Example
 *
 *   parse_str('?type[]=ready_made&type[]=flop', $inputQuery);
 *   $outputQuery = array();
 *
 *   setParam($inputQuery, $outputQuery, 'type', 'ready_made', 'ready');
 *   setParam($inputQuery, $outputQuery, 'type', 'flop');
 * 
 *   // $outputQuery === array(
 *   //   'ready' => 'on',
 *   //   'flop' => 'on'
 *   // );
 *
 */
private function transformQuery($inputQuery, &$outputQuery, $inputParam, $inputValue, $outputParam = null) {
	$outputParam = (is_null($outputParam)) ? $inputValue : $outputParam;

	if(isset($inputQuery[$inputParam]) && in_array($value, $inputQuery[$inputParam])) {
		$outputQuery[$outputParam] = 'on';
	}
}
```